backend/dataall/modules/s3_datasets/db/dataset_repositories.py

```diff
@@ -183,6 +189,20 @@ def get_dataset_tables(session, dataset_uri):
         """return the dataset tables"""
         return session.query(DatasetTable).filter(DatasetTable.datasetUri == dataset_uri).all()
 
+    @staticmethod
+    def update_dataset_shares_expiration(session, dataset):
+        """
+        When share expiration is enabled on the dataset while editing a dataset
+        update all the shares on that dataset and set minimum expiration on them
+        """
+        shares = session.query(ShareObject).filter(ShareObject.datasetUri == dataset.datasetUri).all()
+        for share in shares:
+            share.expiryDate = ShareObjectService.calculate_expiry_date(
```

Comment:
service layer is called from repository layer -- it violates the structure of the code and can lead to nasty circular dependencies. Could you please untangle it?
I believe the logic part can be moved to dataset_service and we only leave actual object update in repository file.

Reply:
Oh Thanks @SofiaSazonova. Since it is a share expiration value I wanted it to be related in the share_base folder, and related services. 
How about if I place it in `share_object_repositories` ?

Reply:
If you touch only share (not the Dataset) -- it belongs to the ShareObjectService (logic part) and ShareObjectRepository(DB update part).

Reply:
In Dataset Service 
before the line 264
` DatasetRepository.update_dataset_shares_expiration(session=session, dataset=dataset)`

I would put calculate_expiry_date  logic -- it has nothing specific to share or any other object, just calculation of the dstes , so we can place it in this Service.
It shouldn't be part of other service to avoid any additional dependencies

and then `ShareObjectRepository.update_dataset_shares_expiration(session=session, dataset_uri=dataset, expiration = dataset.enableExpiration, endDate)`

With this solution we 1) detangle services and repos 2) keep logic in services, db updates in repos, 3) keep share object operations in ShareObjectRepository

Reply:
Hi @SofiaSazonova , Thanks! I think this makes sense.